VERSIONING.md

```diff
@@ -0,0 +1,55 @@
+Rocket Chip Versioning Strategy ** All Strawman to be discussed**
+===============================
+
+Version Naming
+--------------
+
+* Rocket Chip uses a Calendar Versioning release strategy (https://calver.org/).
+* The format is YYYY.MM.DD.MINOR
+* MINOR changes include:
+  * API Additions
+  * Bug Fixes (which can result in changes in the emitted hardware considered bugs)
+* MINOR changes do NOT include:
+  * Changes in top-level I/Os for < level of hierarchy >
+  * Backwards incompatible changes in the API (see below)
+
+API
+---
+
+The Rocket-Chip API is: 
+  * All public Scala methods and classes
+  * All public Scala methods and classes of included Scala libraries (hardfloat, CDE, etc)
+  * The command line used to invoke the generator
+  * All Verilog interfaces (I/Os and parameterization) for verilog files
+  * C code...?
+  * The version(s) of Chisel, FIRRTL, rocket-tools pointed to by rocket-chip.
+  * Any code that generates IO(s) of any Module
+  * The cmd line interface to scripts like vlsi-mem-gen
+  * `public val`s within `LazyModule`s which translate to diplomatic nodes (?? -- this could go either way)
+
+The following is NOT part of the API:
+  * `public val`s within `Module`s which translate to Wires, Regs, module instances
+    * (We note that this may cause issues for people using Cross-Module references)
+  * Internals of scripts like vlsi-mem-gen, the contents of such scripts
+  * Internals of any verilog black boxes
+  * C Code ...?
+      
+Roadmap
+-------
+
+* Upcoming Versions:
+
+  - Version ~2020.04.04.0: Released approximately before the breaking changes discussed for version NEXT:
+  - Version NEXT YY.MM.DD.0: To be released after:
+    - Chisel is bumped to 3.3
+    - FIRRTL is bumped to X.XX
+    - Command line interface updated for staged generator
+    
+* Following versions released quarterly
```

Comment:
If we're planning to do quarterly releases, is it necessary to include the day of the month in the version string? I have doubts to our ability to predict the exact day on which we will cut a release, and I'm not sure if that level of granularity warrants inclusion in the version string.

Reply:
I have zero intention of "predicting" the exact day we will cut a release. I think we should tag it when it happens
However I also think it's quite likely we will cut a release multiple times in a month (hoping that the features listed above happen before the end of April, as a very real example).

So maybe the bug in this section is really "following versions released quarterly" if the releases are going to really be feature-driven

Reply:
> However at the current rate of change it's quite likely we will cut a release multiple times in a month (hoping that the features listed above happen before the end of April, as a very real example, though perhaps would stabilize after that?)

That was not the impression that I was getting from the document, since the only release cadence that was explicitly described was quarterly. Maybe the disconnect is that I was under the impression that SiFive internally would still be using SNAPSHOT/bleeding-edge releases, but that anyone (externally or internally) would have the option of using a more stable release created quarterly (and perhaps with bugfixes). There's always going to be a bit of a tradeoff between stability and new features, and that's why many large projects end up having different channels to cater to different needs.

Reply:
I'm coming around to agree it's pointless to put the day in there. 

Also, i think the concept of a "stable branch" is useful even if releases aren't getting cut from it per se.

Reply:
i removed the day from the propsoed date string